### PR TITLE
Implement the NestedControllerManager controller

### PR DESCRIPTION
**Pull request by charleszheng44**
This PR includes the implementation for the NestedControllerManager

### CODE REVIEW

**Review by Fei-Guo — COMMENTED** [review]
LGTM

**Comment by nikhita** [comment]
creating a (dummy) approve comment to re-trigger CI, the approval mechanism was broken for a bit - kubernetes/test-infra#21687

/approve

File: controllers/controlplane/controller_util.go [hunk 1]
```diff
@@ -60,30 +60,32 @@ func createNestedComponentSts(ctx context.Context,
...
-				"NestedEtcd Statefulset object: %v", err)
+			return fmt.Errorf("fail to generate the Statefulset object: %v", err)
 		}
 
-		ncSvc, err = genServiceObject(ncMeta, ncSpec, ncKind, clusterName, log)
-		if err != nil {
-			return fmt.Errorf("fail to generate the "+
-				"NesteEtcd Service object: %v", err)
+		if ncKind != clusterv1.ControllerManager {
+			// no need to create the service for the NestedControllerManager
```
Comment (christopherhein) [inline controllers/controlplane/controller_util.go:67]: Just a note: For now I think this is true but if we want to run things like metrics service and be able to expose these we'd want the capability later.

**Comment by christopherhein** [comment]
Just the one nit about copyright, lgtm otherwise

File: controllers/controlplane/controller_util.go (hunk 1 above)
Comment (charleszheng44) [reply controllers/controlplane/controller_util.go:67]: Yes, I agree. We can add it back once we need to access the controller manager

**Pull request merged**